### ISSUE TITLE
better aeld event defuser

### ISSUE THREAD

**Issue by anonghuser** [uBlockOrigin/uBlock-issues#3491]
### Prerequisites

- [X] I verified that this is not a filter list issue. Report any issues with filter lists or broken website functionality in the [uAssets issue tracker](https://github.com/uBlockOrigin/uAssets/issues).
- [X] This is **NOT** a [_YouTube_](https://www.github.com/uBlockOrigin/uAssets/issues/20586), [_Facebook_](https://www.github.com/uBlockOrigin/uAssets/issues/3367) or [_Twitch_](https://www.github.com/uBlockOrigin/uAssets/issues/5184) report. _These sites MUST be reported by clicking their **respective links**_.
- [X] This is not a support issue or a question. For support, questions, or help, visit [/r/uBlockOrigin](https://www.reddit.com/r/uBlockOrigin/).
- [X] I performed a [cursory search of the issue tracker](https://github.com/uBlockOrigin/uBlock-issues/issues?q=is%3Aissue) to avoid opening a duplicate issue.
- [X] The issue is not present after disabling uBO in the browser.
- [X] I checked the [documentation](https://github.com/gorhill/uBlock/wiki) to understand that the issue I am reporting is not normal behavior.

### I tried to reproduce the issue when...

- [X] uBO is the only extension.
- [X] uBO uses default lists and settings.
- [X] using a new, unmodified browser profile.

### Description

currently, the `aeld` scriptlet does not stop events set with `onxxxx` attribute/property
can we please get another version or a flag which instead of monkey-patching addEventListener just sets a capture phase event handler on the window which calls stopImmediatePropagation to block all other handlers no matter how they are set?
if it is the first such handler that is set it is guaranteed to be the first executed too, so seems like a more reliable approach, except for the few rare events that don't bubble.

### A specific URL where the issue occurs.

```yaml
-
```

### Steps to Reproduce

-

### Expected behavior

-

### Actual behavior

-

### uBO version

-

### Browser name and version

-

### Operating System and version

-

**Comment by gorhill** [comment]
Provide actual real world cases please, I prefer to not spend time on code which has no real world use case -- that is why we ask for _"specific URL where the issue occurs"_.

**Comment by krystian3w** [comment]
To me, the only thing that comes to mind is masking events with e.g. jQuery, but that's something even AdGuard developers don't want to touch.

https://github.com/AdguardTeam/Scriptlets/issues/362#issuecomment-1739677779

In addition, for uBO perhaps a selector (https://github.com/gorhill/uBlock/commit/060f9d68fc641d08642a42a3903ccc89634735dc) will also help for such accidents:

> ```adb
> #%#//scriptlet('prevent-addEventListener', 'click', '/window.open|adsdomain.com|countdownDownload|btn-download/', 'elements', '#btn-download')
> ```

Unfortunately, that request there is also a PoC and not a real threat, where we need to find another method to neutralize the problem.

**Comment by anonghuser** [comment]
hi, @krystian3w @gorhill 

a sample url of my specific issue is https://www.foxteller.com/novel/643/1/the-witch-and-the-mercenary/001
i am actually able to remove the oncopy/oncut/oncontextmenu attributes with i.e. `[nocopy]:remove-attr(nocopy)` but the console keeps showing event listeners on the element - without looking at the site's script my guess is it sets them as properties, not just as attributes

i didn't provide the specifics since i didn't want to turn this into a support request, just a generally useful feature request. the approach with capture phase stopImmediatePropagation works rather well from my experience - i actually have a userscript for it for the contextmenu event. i would have solved the copy/cut with a userscript as well if it bothered me too much, but uBO was more convenient for a few other annoyances on the site - css for unstickying the header and allowing user selection, disabling console.log since they spam it, etc., so i thought i'd try to tack on the copy/cut things to that too.

**Comment by stephenhawk8054** [comment]
Which browser are you using?

**Comment by anonghuser** [comment]
edge

**Comment by stephenhawk8054** [comment]
`foxteller.com##[oncontextmenu]:remove-attr(/oncontextmenu|oncopy|oncut/)` works better on Firefox. For chromium, you'll need

```adb
foxteller.com##+js(ra, oncontextmenu|oncopy|oncut, [id^="chapter"], stay)
```

> Browser name and version

That's why please always provide all information in the template, even for Feature request. Different browsers can have different behaviors.

**Comment by anonghuser** [comment]
indeed, that solved the issue, thanks. i didn't try it since the docs were suggesting the newer approach.
